openam-shared/src/main/java/com/sun/identity/shared/debug/format/impl/DebugFormatterFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The contents of this file are subject to the terms of the Common Development and
+ * Distribution License (the License). You may not use this file except in compliance with the
+ * License.
+ *
+ * You can obtain a copy of the License at legal/CDDLv1.1.txt. See the License for the
+ * specific language governing permission and limitations under the License.
+ *
+ * When distributing Covered Software, include this CDDL Header Notice in each file and include
+ * the License file at legal/CDDLv1.1.txt. If applicable, add the following below the CDDL
+ * Header, with the fields enclosed by brackets [] replaced by your own identifying
+ * information: "Portions copyright [year] [name of copyright owner]".
+ *
+ * Copyright 2025 Wren Security. All rights reserved.
+ */
+package com.sun.identity.shared.debug.format.impl;
+
+import com.sun.identity.shared.configuration.SystemPropertiesManager;
+import com.sun.identity.shared.debug.DebugConstants;
+import com.sun.identity.shared.debug.format.DebugFormatter;
+
+/**
+ * Expose a singleton {@link DebugFormatter} determined by the
+ * {@value DebugConstants#CONFIG_DEBUG_LOGFILE_FORMAT} system property.
+ */
+public class DebugFormatterFactory {
+
+    private static final DebugFormatter formatterInstance;
```

Comment:
We might want to be able to respond to server configuration changes during boot and runtime. So maybe we should have `resetInstance` method that will reinitialize the formatter from `Debug#resetDebug` (this will allow specifying formatter in persisted server properties).

Reply:
Good point. I’ve implemented the `resetInstance` method and the necessary updates.